Pomodoro/Sources/App/ViewController.swift

```diff
@@ -6,15 +6,243 @@
+            endAngle: 270.degreesToRadians,
+            clockwise: true
+        ).cgPath
+        foregroundProgressLayer.strokeColor = animationColor.cgColor
+        foregroundProgressLayer.fillColor = UIColor.clear.cgColor
+        foregroundProgressLayer.lineWidth = 8
+        view.layer.addSublayer(foregroundProgressLayer)
+    }
+
+    private func startResumeAnimation() {
+        !isAnimationStarted ? startAnimation() : resumeAnimation()
+    }
+
+    private func startAnimation() {
+        resetAnimation()
+        foregroundProgressLayer.strokeEnd = 0.0
+        animation.keyPath = "strokeEnd"
+        animation.fromValue = 0
+        animation.toValue = 1
+        animation.duration = Double(time)
+        animation.delegate = self
+        animation.isRemovedOnCompletion = false
+        animation.isAdditive = true
+        animation.fillMode = CAMediaTimingFillMode.forwards
+        foregroundProgressLayer.add(animation, forKey: "strokeEnd")
+        isAnimationStarted = true
+    }
+
+    private func resetAnimation() {
+        foregroundProgressLayer.speed = 1.0
+        foregroundProgressLayer.timeOffset = 0.0
+        foregroundProgressLayer.beginTime = 0.0
+        foregroundProgressLayer.strokeEnd = 0.0
+        isAnimationStarted = false
+    }
+
+    private func pauseAnimation() {
+        let pausedTime = foregroundProgressLayer.convertTime(CACurrentMediaTime(), from: nil)
+        foregroundProgressLayer.speed = 0.0
+        foregroundProgressLayer.timeOffset = pausedTime
+    }
+
+    private func resumeAnimation() {
+        let pausedTime = foregroundProgressLayer.timeOffset
+        foregroundProgressLayer.speed = 1.0
+        foregroundProgressLayer.timeOffset = 0.0
+        foregroundProgressLayer.beginTime = 0.0
+        let timeSincePaused = foregroundProgressLayer.convertTime(CACurrentMediaTime(), from: nil) - pausedTime
+        foregroundProgressLayer.beginTime = timeSincePaused
     }
 
+    private func stopAnimation() {
+        foregroundProgressLayer.speed = 1.0
+        foregroundProgressLayer.timeOffset = 0.0
+        foregroundProgressLayer.beginTime = 0.0
+        foregroundProgressLayer.strokeEnd = 0.0
+        foregroundProgressLayer.removeAllAnimations()
+        isAnimationStarted = false
+    }
+
+    private func startTimer() {
+        timer = Timer.scheduledTimer(
+            timeInterval: 1,
+            target: self,
+            selector: #selector(updateTimer),
+            userInfo: nil,
+            repeats: true
+        )
+    }
+
+    @objc private func updateTimer() {
+        if time <= 1 {
+            modeLabel.text = isWorkTime ? "Rest" : "Work"
+            timer.invalidate()
+            startPauseButton.setImage(UIImage(systemName: "play", withConfiguration: largeButtonConfig)?
+                .withTintColor(.red, renderingMode: .alwaysOriginal), for: .normal)
+
+            stopAnimation()
+            isWorkTime = !isWorkTime
+            time = isWorkTime ? 25 : 5
```

Comment:
it is preferred to not write magic numbers. Better create two constants like:
private let workTime = 25
private let restTime = 5

and use only these constants like -> time = isWorkTime ? workTime : restTime